DbAccessor/src/Messages/GetAllConversations.cs

```diff
@@ -0,0 +1,87 @@
+using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging;
+using MongoDB.Driver;
+using MongoDB.Bson;
+using Microsoft.Azure.Functions.Worker;
+
+namespace GoRideShare
+{
+    public class GetAllConversations
+    {
+
+        private readonly string? _baseApiUrl;
+        private readonly ILogger<GetAllConversations> _logger;
+        // initialize the MongoDB client lazily. This is a best practice for serverless functions because it is not efficient to establish Mongo connections on every execution of our Azure Function
+        public static Lazy<MongoClient> lazyClient = new Lazy<MongoClient>(InitializeMongoClient);
+        public static MongoClient client = lazyClient.Value;
+
+        public static MongoClient InitializeMongoClient()
+        {
+            return new MongoClient(Environment.GetEnvironmentVariable("MONGODB_ATLAS_URI"));
+        }
+
+        public GetAllConversations(ILogger<GetAllConversations> logger)
+        {
+            _logger = logger;
+            _baseApiUrl = Environment.GetEnvironmentVariable("BASE_API_URL");
+        }
+
+        [Function("GetAllConversations")]
+        public async Task<IActionResult> Run([HttpTrigger(AuthorizationLevel.Anonymous, "get")] HttpRequest req)
+        {
+            // If validation result is not null, return the bad request result
+            var validationResult = Utilities.ValidateHeaders(req.Headers, out string userId);
+            if (validationResult != null)
+            {
+                return validationResult;
+            }
+
+
+            // Get the user name and photo details by calling SQL Get user endpoint
+            User otherUser;
+            string endpoint = $"{_baseApiUrl}/api/GetUser";
+
+            // make http request using req.Headers and endpoint to get the user details
+            var (error, response) = await Utilities.MakeHttpGetRequest(userId, endpoint);
```

Comment:
This is sending a request back to the logic layer?

Reply:
We should be able to just get the user directly from the db?

Reply:
Ideally we should get it directly without http requests. Variables are not named correctly here, but its pointing at DB Layer's GetUser endpoint. I will fix it